packages/ui-extensions-react/src/surfaces/checkout/errors.ts

```diff
@@ -18,6 +18,16 @@ export class ExtensionHasNoMethodError extends Error {
   }
 }
 
+export class ExtensionHasNoFieldError extends Error {
```

Comment:
A bunch of our hooks only work on the ODP. We're throwing this error when they are used outside of the ODP.

This is done so that we can keep the typings that we have. Alternatively, we'd have to mark all of them as optional which would be a breaking change for existing OSP/ODP extensions.